packages/catalog-search/package.json

```diff
@@ -45,27 +45,27 @@
   "devDependencies": {
     "@edx/browserslist-config": "1.1.0",
     "@edx/frontend-build": "12.7.0",
-    "@edx/frontend-platform": "4.0.1",
+    "@edx/frontend-platform": "4.6.0",
     "@edx/paragon": "19.25.1",
     "@fortawesome/free-solid-svg-icons": "5.8.1",
-    "@fortawesome/react-fontawesome": "0.1.4",
+    "@fortawesome/react-fontawesome": "0.2.0",
     "@testing-library/jest-dom": "5.11.6",
-    "@testing-library/react": "11.2.6",
+    "@testing-library/react": "12.1.4",
     "@testing-library/react-hooks": "3.4.2",
     "@testing-library/user-event": "13.1.8",
-    "react": "16.14.0",
-    "react-dom": "16.14.0",
+    "react": "17.0.2",
+    "react-dom": "17.0.2",
     "react-instantsearch-dom": "6.8.3",
     "react-router-dom": "5.2.0",
-    "react-test-renderer": "16.14.0"
+    "react-test-renderer": "17.0.2"
   },
   "peerDependencies": {
     "@edx/frontend-platform": "^4.0.1",
     "@edx/paragon": "^19.15.0 || ^20.0.0",
     "@fortawesome/free-solid-svg-icons": "^5.8.1",
-    "@fortawesome/react-fontawesome": "^0.1.4",
-    "react": "^16.12.0",
-    "react-dom": "^16.12.0",
+    "@fortawesome/react-fontawesome": "^0.2.0",
+    "react": "^16.12.0 || ^17.0.0",
+    "react-dom": "^16.12.0 || ^17.0.0",
```

Comment:
nit: could opt for `^17` if we want to allow all major versions of React v17. Shorthand for `^17.0.0`.

Reply:
I kind of followed the convention from other MFEs (paragon for example) also its more explicit, but since both ^17.0.0 and ^17 are functionally equivalent I can use the shorthand for it if that is preferred.